### PR TITLE
Add more information to `FromRowError`

### PR DESCRIPTION
**Pull request by havaker**
`FromRowError::BadCqlVal` was made more useful by adding a related column index.
Also, error messages were changed from "Bad CQL Value" to something like "Value is null in the column with index 2".

- [x] I have split my patch into logically separate commits.
- [x] All commit messages clearly explain what they change and why.
- [x] I added relevant tests for new features and bug fixes.
- [x] All commits compile, pass static checks and pass test. <- clippy changed
- [x] PR description sums up the changes and reasons why they should be introduced.
- [x] I added appropriate `Fixes:` annotations to PR description.

Fixes: #287

### CODE REVIEW

File: scylla-macros/src/from_row.rs
Comment (piodul) [inline scylla-macros/src/from_row.rs:26]: You can use `map_err` here:

```rust
<#field_type as FromCqlVal<Option<CqlValue>>>::from_cql(col_value)
    .map_err(|e| FromRowError::BadCqlVal {
        err: e,
        column: col_ix,
    })?
```

File: scylla/src/frame/response/cql_to_rust.rs
Comment (piodul) [inline scylla/src/frame/response/cql_to_rust.rs:139]: Use `map_err` here, too

**Review by piodul — CHANGES_REQUESTED** [review]
I know that the original issue mentioned adding information about the column which failed to parse - however I think that we can improve `FromRowError` even further while we are at it.

We should also improve the `RowTooShort` variant. There are two things I think should be done with it:

- Change the name to `WrongRowSize` and return it also in the case when the row is too big (now I think we are just truncating the row, which is not what we want) - some tests for too large rows would be nice
- Include information about expected row size vs. actual row size

File: scylla-macros/src/from_row.rs
Comment (havaker) [reply scylla-macros/src/from_row.rs]: Done

File: scylla/src/frame/response/cql_to_rust.rs
Comment (havaker) [reply scylla/src/frame/response/cql_to_rust.rs]: Done

**Comment by havaker** [comment]
v2: applied suggestions from @piodul's review, improved the `RowTooShort` variant (by renaming it and adding expected and actual row count)

**Comment by psarna** [comment]
Note: I pushed your clippy patch separately, since it's not directly related and at the same time needed to unblock CI, please rebase.

File: scylla-macros/src/from_row.rs
Comment (piodul) [inline scylla-macros/src/from_row.rs:42]: Nitpick: please put `expected` and `actual` field into separate lines (like `cargo fmt` would if it knew how to format macros).

**Review by piodul — APPROVED** [review]
Left some nitpicks, otherwise looks good. Please rebase according to @psarna 's request.

File: scylla/src/frame/response/cql_to_rust.rs
Comment (havaker) [reply scylla/src/frame/response/cql_to_rust.rs]: Done

File: scylla/src/frame/response/cql_to_rust.rs
Comment (havaker) [reply scylla/src/frame/response/cql_to_rust.rs]: Done

File: scylla-macros/src/from_row.rs
Comment (havaker) [reply scylla-macros/src/from_row.rs]: Done

**Comment by havaker** [comment]
v3: rebased, applied @piodul's nitpicks

**Review by psarna — APPROVED** [review]
@piodul let me know if you approve of the fixes for your comments, or just go ahead and click the merge button if you do

**Comment by psarna** [comment]
@piodul ping, I just need a quick ack to merge

**Review by piodul — APPROVED** [review]

**Review by psarna — APPROVED** [review]

**Pull request merged**
